packages/contracts/test/DepositFeed.ts

```diff
@@ -0,0 +1,155 @@
+  })
+  it('Should revert if a contract creation has a non-zero destination address', async () => {
+    await expect(
+      depositFeed.depositTransaction(
+        NON_ZERO_ADDRESS,
+        NON_ZERO_GASLIMIT,
+        true,
+        '0x'
+      )
+    ).to.be.revertedWith(
+      'Contract creation deposits must not specify a recipient address.'
+    )
+  })
+
+  describe('Should emit the correct log values when:', async () => {
+    it('a simple ETH-only transaction is deposited.', async () => {
+      const tx = await depositFeed.depositTransaction(
+        NON_ZERO_ADDRESS,
+        NON_ZERO_GASLIMIT,
+        false,
+        '0x',
+        {
+          value: NON_ZERO_VALUE,
+        }
+      )
+
+      const event = decodeDepositEvent((await tx.wait()).logs[0])
+
+      expect(event).deep.equal({
+        from: signerAddress,
+        to: NON_ZERO_ADDRESS,
+        value: NON_ZERO_VALUE,
+        gasLimit: NON_ZERO_GASLIMIT,
+        isCreation: false,
+        data: '0x',
+      })
+    })
+
+    it('a simple contract creation is deposited with 0 value.', async () => {
+      const tx = await depositFeed.depositTransaction(
+        ZERO_ADDRESS,
+        NON_ZERO_GASLIMIT,
+        true,
+        '0x'
+      )
+
+      const event = decodeDepositEvent((await tx.wait()).logs[0])
+
+      expect(event).deep.equal({
+        from: signerAddress,
+        to: ZERO_ADDRESS,
+        value: ZERO_BIGNUMBER,
+        gasLimit: NON_ZERO_GASLIMIT,
+        isCreation: true,
+        data: '0x',
+      })
+    })
+
+    it('a simple contract creation is deposited with some eth value.', async () => {
+      const tx = await depositFeed.depositTransaction(
+        ZERO_ADDRESS,
+        NON_ZERO_GASLIMIT,
+        true,
+        '0x',
+        {
+          value: NON_ZERO_VALUE,
+        }
+      )
+
+      const event = decodeDepositEvent((await tx.wait()).logs[0])
+
+      expect(event).deep.equal({
+        from: signerAddress,
+        to: ZERO_ADDRESS,
+        value: NON_ZERO_VALUE,
+        gasLimit: NON_ZERO_GASLIMIT,
+        isCreation: true,
+        data: '0x',
+      })
+    })
+
+    it('a contract creation is deposited from a contract.', async () => {
+      // Deploy a dummy contract so we can impersonate it
+      const Dummy = await ethers.getContractFactory('Dummy')
+      const dummy = await Dummy.deploy()
+      await dummy.deployed()
+
+      const tx = await dummy.forward(
+        depositFeed.address,
+        depositFeed.interface.encodeFunctionData('depositTransaction', [
+          ZERO_ADDRESS,
+          NON_ZERO_GASLIMIT,
+          true,
+          NON_ZERO_DATA,
+        ])
+      )
+      const event = decodeDepositEvent((await tx.wait()).logs[0])
+
+      expect(event).deep.equal({
+        from: applyL1ToL2Alias(dummy.address),
+        to: ZERO_ADDRESS,
+        value: ZERO_BIGNUMBER,
+        gasLimit: NON_ZERO_GASLIMIT,
+        isCreation: true,
+        data: NON_ZERO_DATA,
+      })
+    })
+  })
+})
```

Comment:
Do we want to use a solidity code coverage tool?

Reply:
Sure, I have some experience [setting it up](https://github.com/ethereum-optimism/optimism/commit/46d30311d69af4a7f757bc9ec1863d4bf90442fd). I'd say it's a nice to have though considering the size of the code. Created the issue here: #80